kogito-editors-js/packages/feel-input-component/src/__tests__/Monaco/FeelConfigs.test.ts

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  feelTheme,
+  feelTokensConfig,
+  feelDefaultConfig,
+  feelDefaultSuggestions,
+  MONACO_FEEL_LANGUAGE,
+  MONACO_FEEL_THEME,
+} from "../../";
+
+describe("FeelConfigs", () => {
+  test("feelTheme", () => {
+    const theme = feelTheme();
+
+    expect(theme.base).toBe("vs");
+    expect(theme.inherit).toBeTruthy();
+    expect(theme.rules).toHaveLength(5);
+    expect(theme.colors["editorLineNumber.foreground"]).toBe("#000000");
+  });
+
+  test("feelTokensConfig", () => {
+    const tokens = feelTokensConfig().tokenizer.root;
+
+    expect(tokens).toHaveLength(6);
+    expect(tokens.map((t: any) => t[1])).toEqual([
+      "feel-boolean",
+      "feel-numeric",
+      "feel-string",
+      "feel-function",
+      "feel-keyword",
+      "feel-keyword",
+    ]);
+  });
+
+  test("feelDefaultConfig", () => {
+    const value = "test";
+    const config = feelDefaultConfig({ value });
+
+    expect(config.language).toBe(MONACO_FEEL_LANGUAGE);
+    expect(config.theme).toBe(MONACO_FEEL_THEME);
+    expect(config.fontSize).toBe(15);
+    expect(config.contextmenu).toBe(false);
+    expect(config.useTabStops).toBe(false);
+    expect(config.folding).toBe(false);
+    expect(config.automaticLayout).toBe(true);
+    expect(config.lineNumbersMinChars).toBe(0);
+    expect(config.overviewRulerBorder).toBe(false);
+    expect(config.scrollBeyondLastLine).toBe(false);
+    expect(config.hideCursorInOverviewRuler).toBe(true);
+    expect(config.scrollbar?.useShadows).toBe(false);
+    expect(config.minimap?.enabled).toBe(false);
+    expect(config.value).toBe(value);
+  });
+
+  test("feelDefaultSuggestions", () => {
```

Comment:
if I compare the number of entries in `FeelConfig.ts` and the number of entries we cover by test here, I need to say we do not cover them all. How we are sure we configured monaco for all supported keywords, functions etc?

do we have some service that returns list of reserved keywords? shouldn't we compare number of entries returned by such a serive with number of entries in `FeelConfig.ts`?

Reply:
@jomarko In the test assertions, we're checking if we have 47 entries, and we're also checking 2 samples of each type of suggestion. In other words, 2 samples of keyword suggestions and 2 samples of function suggestions.

With these assertions, I believe we already have a high level of confidence without too much duplication. However, if you think it's a good idea to add assertions for all 47 entries, we can add them. I don't think it would be harmful, but I personally think it would be a bit of duplication.